grpc/server/src/plugins/action/actionrpc_impl.h

```diff
@@ -36,13 +33,12 @@ class ActionServiceImpl final : public rpc::action::ActionService::Service
     Status Land(ServerContext *context, const rpc::action::LandRequest *request,
                 rpc::action::ActionResult *response) override
     {
-        const Action::Result action_result = action->land();
+        const Action::Result action_result = action.land();
         response->set_result(static_cast<rpc::action::ActionResult::Result>(action_result));
         response->set_result_str(Action::result_str(action_result));
         return Status::OK;
     }
 
 private:
-    DroneCore *dc;
-    std::shared_ptr<Action> action;
+    const Action &action;
```

Comment:
If its `const`, then how does `action.arm()` can work ? IMO, `const` is used when you do **read-only**. Operations that change the state (such as arm, takeoff, etc) can't be `const`. Please correct me if I am wrong.

Reply:
The `Action` object doesn't have state. It just passes the message further. By having it `const`, it makes it clear. `Mission` has some internal state and, therefore, isn't const.

Reply:
I see.